api/src/routes/api.ts

```diff
@@ -6,11 +6,13 @@ import Config from "../models/Config";
 import Job from "../models/Job";
 
 const router = express.Router();
+const { body, validationResult } = require("express-validator");
 import { setupPassport, requireToken } from "./auth";
+import fs = require("fs");
 setupPassport(router);
 
 function getJobFromRequest(req): Job {
-    // TODO: sanitize parameters!
+    //TODO make sure Job exists
```

Comment:
It should be possible to do this very similarly to the check in the /:category route. You just need to see if the job directory (`holdingArea() + req.params.category + "/" + req.params.job`) actually exists. If it doesn't, you need to figure out how to trigger a 404 error. Perhaps you could return null from this function, and then you have to check the return value in all the places where this gets called. Maybe there's an easier way, like creating an "addJobToRequest" callback that you add to the chain in the route, that creates a this.request.job based on the other request parameters -- then the final callback in each chain could use this.request.job instead of calling getJobFromRequest, and this function could just return the 404 and break the chain if the file is not found.

Reply:
Updated!